src/content/app/genome-browser/state/browser-sidebar-modal/browserSidebarModalSlice.ts

```diff
@@ -0,0 +1,289 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PayloadAction } from '@reduxjs/toolkit';
+import { ThunkAction } from 'redux-thunk';
+import { createSlice } from '@reduxjs/toolkit';
+import { Action } from 'redux';
+import pick from 'lodash/pick';
+
+import browserSidebarModalStorageService from 'src/content/app/genome-browser/components/browser-sidebar-modal/services/browser-sidebar-modal-storage-service';
+import browserStorageService from 'src/content/app/genome-browser/services/browser-storage-service';
+
+import {
+  getBrowserActiveGenomeId,
+  getBrowserActiveFocusObject
+} from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
+import {
+  getActiveGenomePreviouslyViewedObjects,
+  getActiveBrowserSidebarModal
+} from './browserSidebarModalSelectors';
+
+import { ParsedUrlPayload } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSlice';
+import { RootState } from 'src/store';
+
+export type PreviouslyViewedObject = {
+  genome_id: string;
+  object_id: string;
+  type: string;
+  label: string | string[];
+};
+
+export enum BrowserSidebarModalView {
+  SEARCH = 'search',
+  TRACKS_MANAGER = 'Tracks manager',
+  BOOKMARKS = 'Previously viewed',
+  PERSONAL_DATA = 'Personal data',
+  SHARE = 'Share',
+  DOWNLOADS = 'Downlods'
+}
+
+export type PreviouslyViewedObjects = {
+  [genomeId: string]: PreviouslyViewedObject[];
+};
+
+export type BrowserSidebarModalStateForGenome = Readonly<{
+  isBrowserSidebarModalOpened: boolean;
+  browserSidebarModalView: BrowserSidebarModalView | null;
+  bookmarks: PreviouslyViewedObject[];
+  previouslyViewedObjects: PreviouslyViewedObject[];
+}>;
+
+export type BrowserSidebarModalState = Readonly<{
+  [genomeId: string]: BrowserSidebarModalStateForGenome;
+}>;
+
+export const defaultBrowserSidebarModalStateForGenome: BrowserSidebarModalStateForGenome =
+  {
+    isBrowserSidebarModalOpened: false,
```

Comment:
Is there any point in having this key to the state at all? Isn't `browserSidebarModalView` key sufficient to indicate whether the modal is open (it should be if it's not null)?

Reply:
✅